src/Workspaces/CSharp/Portable/Extensions/ContextQuery/SyntaxTreeExtensions.cs

```diff
@@ -2531,7 +2531,7 @@ public static bool IsNameOfContext(this SyntaxTree syntaxTree, int position, Sem
             return false;
         }
 
-        public static bool IsIsOrAsContext(this SyntaxTree syntaxTree, int position, SyntaxToken tokenOnLeftOfPosition, CancellationToken cancellationToken)
+        public static bool IsIsOrAsOrSwitchExpressionContext(this SyntaxTree syntaxTree, int position, SyntaxToken tokenOnLeftOfPosition, CancellationToken cancellationToken)
```

Comment:
this method looks pretty busted in many ways.  For example:

```c#
 if (token.GetAncestor<BlockSyntax>() == null)
```

That seems like it means that we won't get 'switch' completion in an arrow method.  Can you add test for this and fix up this method accordingly.

Note: afaict this method could be so much simpler.  It's really just "are we after a complete expression"

Reply:
nit: consider renameing "IsAfterCompletelyWrittenExpression".  but i don't care :)